collect_app/src/androidTest/java/org/odk/collect/android/utilities/DateTimeUtilsTest.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2017 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.odk.collect.android.utilities;
+
+import android.content.Context;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.logic.DatePickerDetails;
+
+import java.util.Date;
+import java.util.Locale;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(AndroidJUnit4.class)
+public class DateTimeUtilsTest {
+
+    private DatePickerDetails gregorianDatePickerDetails;
+    private DatePickerDetails ethiopianDatePickerDetails;
+
+    private Context context;
+
+    @Before
+    public void setUp() {
+        gregorianDatePickerDetails = new DatePickerDetails(DatePickerDetails.DatePickerType.GREGORIAN, DatePickerDetails.DatePickerMode.CALENDAR);
+        ethiopianDatePickerDetails = new DatePickerDetails(DatePickerDetails.DatePickerType.ETHIOPIAN, DatePickerDetails.DatePickerMode.SPINNERS);
+
+        context = Collect.getInstance();
+    }
+
+    @Test
+    public void getDateTimeLabelTest() {
+        long dateInMilliseconds = 687967200000L; // 20 Oct 1991 14:00
+        Locale.setDefault(new Locale("af"));
```

Comment:
Are all of these different locale tests actually testing different cases? I think one Gregorian English, one Gregorian non-English, one Ethiopic Amharic and one Ethiopic English would be enough. This would show that dates are localized and that in the case of an Ethiopic date the Gregorian date in parens is always in English.